Loop/Managers/LoopAlertsManager.swift

```diff
@@ -50,13 +60,104 @@ public class LoopAlertsManager {
+                notification.body = String(format: NSLocalizedString("Loop has not completed successfully in %@", comment: "The notification alert describing a long-lasting loop failure. The substitution parameter is the time interval since the last loop"), failureIntervalString)
+            }
+
+            notification.title = NSLocalizedString("Loop Failure", comment: "The notification title for a loop failure")
+            if isCritical, FeatureFlags.criticalAlertsEnabled {
+                if #available(iOS 15.0, *) {
+                    notification.interruptionLevel = .critical
+                }
+                notification.sound = .defaultCritical
+            } else {
+                if #available(iOS 15.0, *) {
+                    notification.interruptionLevel = .timeSensitive
+                }
+                notification.sound = .default
+            }
+            notification.categoryIdentifier = LoopNotificationCategory.loopNotRunning.rawValue
+            notification.threadIdentifier = LoopNotificationCategory.loopNotRunning.rawValue
+
+            let trigger = UNTimeIntervalNotificationTrigger(
+                timeInterval: failureInterval + gracePeriod,
+                repeats: false
+            )
+
+            let request = UNNotificationRequest(
+                identifier: "\(LoopNotificationCategory.loopNotRunning.rawValue)\(failureInterval)",
+                content: notification,
+                trigger: trigger
+            )
+
+            if let nextTriggerDate = trigger.nextTriggerDate() {
+                let scheduledNotification = StoredLoopNotRunningNotification(
+                    alertAt: nextTriggerDate,
+                    title: notification.title,
+                    body: notification.body,
+                    timeInterval: failureInterval,
+                    isCritical: isCritical)
+                scheduledNotifications.append(scheduledNotification)
+            }
+            UNUserNotificationCenter.current().add(request)
+        }
+        UserDefaults.appGroup?.loopNotRunningNotifications = scheduledNotifications
+    }
+
+    func clearLoopNotRunningNotifications(alertStore: AlertStore) {
+
+        // Any past alerts have been delivered at this point
+        let now = Date()
+        for notification in UserDefaults.appGroup?.loopNotRunningNotifications ?? [] {
+            if notification.alertAt < now {
+                let alertIdentifier = Alert.Identifier(managerIdentifier: "Loop", alertIdentifier: "loopNotLooping")
+                let content = Alert.Content(title: notification.title, body: notification.body, acknowledgeActionButtonLabel: "ios-notification-default")
+                let interruptionLevel: Alert.InterruptionLevel = notification.isCritical ? .critical : .timeSensitive
+                let alert = Alert(identifier: alertIdentifier, foregroundContent: nil, backgroundContent: content, trigger: .immediate, interruptionLevel: interruptionLevel)
```

Comment:
I'm fine with this, but it is sort of weird to record, retroactively, an `.immediate` alert here, when really it was a `.delayed()` alert (with some time interval) that had expired.  I don't think it matters at the end of the day, but it is something to think about.

Reply:
Again, I don't feel _that_ strongly... it is just technically "more correct" to store them as `.delayed` because they, technically, are.